### DISCUSSION TITLE
在其上下文中，该请求的地址无效

### DISCUSSION THREAD

**Discussion by CodingMoeButa** [HMBSbige/NatTypeTester#237] [Q&A]
使用V2Ray作为Socks5代理，点击测试，立即弹窗提示“在其上下文中，该请求的地址无效。”

**Comment by HMBSbige** [comment]
一般是 v6 代理，请求了 v4，或者反过来

暂时不支持这种

**Reply by CodingMoeButa** [reply]
不是这种情况。是NatTypeTester无法连上本地由V2Ray开启的Socks代理，跟V2Ray后面的代理服务器都没关系。

**Reply by HMBSbige** [reply]
对啊，就是 v4 socks5 代理，请求了 v6 这种啊
确实跟实际落地没什么关系

举例来说你要测试 v6 的话，`服务器`和`本地地址`都要写 v6 的地址

**Reply by CodingMoeButa** [reply]
但是我没有请求IPv6啊。本地Socks服务器绑定在127.0.0.1:10808，我在NatTypeTester填的代理设置也是127.0.0.1:10808，就报这个错。我都怀疑是不是两者对Socks的实现不一样而造成了兼容性问题。

**Reply by HMBSbige** [reply]
仔细想了想报错，还真可能是

但是本项目 Socks5 实现是符合 RFC1918 的，并且多次检查过

如果真有问题，建议具体指出（因为 v2ray 很复杂

**Reply by CodingMoeButa** [reply]
按理说v2ray-core的Socks实现应该也是符合RFC的。如果要调试的话，我可以把NatTypeTester的源代码拉取下来，在发起测试请求的地方下断点，看看具体是什么情况。不过我不打算弄了，我已经用Netch内置的NAT检测功能看到了结果，这个问题到此为止吧。
